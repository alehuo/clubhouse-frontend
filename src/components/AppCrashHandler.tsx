import React from "react";
import FontAwesome from "react-fontawesome";
import styled from "styled-components";

const bugReportUrl = "https://github.com/alehuo/clubhouse-frontend/issues/new";

const AppCrashScreenWrapper = styled.div`
  height: 100vh;
  width: 100vw;
  color: #ff8484;
  text-align: center;
  padding-top: 50px;
`;

interface State {
  error?: Error;
  errorInfo?: React.ErrorInfo;
}

class AppCrashHandler extends React.Component<{}, State> {
  constructor(props: {}) {
    super(props);
    this.state = { error: undefined, errorInfo: undefined };
  }

  public componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    this.setState({
      error,
      errorInfo,
    });
  }

  public render() {
    if (this.state.errorInfo !== undefined && this.state.error !== undefined) {
      return (
        <AppCrashScreenWrapper>
          <FontAwesome name={"frown"} size={"4x"} />
          <h3>The application has crashed.</h3>
          <p>
            Please report the bug at{" "}
            <a href={bugReportUrl} target="__blank">
              {bugReportUrl}
            </a>
          </p>
          <p>{this.state.error !== undefined && this.state.error}</p>
          <p>{this.state.errorInfo !== undefined && this.state.errorInfo}</p>
        </AppCrashScreenWrapper>
      );
    }
    return this.props.children;
  }
}

export { AppCrashHandler, AppCrashScreenWrapper };
