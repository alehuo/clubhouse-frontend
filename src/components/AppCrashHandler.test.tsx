import { shallow } from "enzyme";
import React from "react";
import { AppCrashHandler, AppCrashScreenWrapper } from "./AppCrashHandler";

it("AppCrashHandler renders without crashing", () => {
  const cmpnt = shallow(<AppCrashHandler />);
  expect(cmpnt).toMatchSnapshot();
});

it("AppCrashScreenWrapper renders without crashing", () => {
  const cmpnt = shallow(<AppCrashScreenWrapper />);
  expect(cmpnt).toMatchSnapshot();
});
