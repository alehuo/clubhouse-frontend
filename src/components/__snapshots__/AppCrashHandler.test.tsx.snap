// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCrashHandler renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppCrashHandler />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AppCrashScreenWrapper renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  height: 100vh;
  width: 100vw;
  color: #ff8484;
  text-align: center;
  padding-top: 50px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  height: 100vh;
  width: 100vw;
  color: #ff8484;
  text-align: center;
  padding-top: 50px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
